backend/hdfs/fs.go

```diff
@@ -35,11 +83,26 @@ func NewFs(ctx context.Context, name, root string, m configmap.Mapper) (fs.Fs, e
 		return nil, err
 	}
 
-	client, err := hdfs.NewClient(hdfs.ClientOptions{
+	options := hdfs.ClientOptions{
 		Addresses:           []string{opt.Namenode},
-		User:                opt.Username,
 		UseDatanodeHostname: false,
-	})
+	}
+
+	if len(opt.ServicePrincipalName) > 0 {
+		options.KerberosClient, err = getKerberosClient()
+		if err != nil {
+			return nil, fmt.Errorf("Problem with kerberos authentication: %s", err)
+		}
+		options.KerberosServicePrincipleName = opt.ServicePrincipalName
```

Comment:
Principle or Principal? We should stick to one spelling!

I think `principal` is correct, which means that `hdfs.ClientOptions` is spelled wrong!

Reply:
yes.

should we replicate tipo from hdfs library ?

Reply:
That is in the exported struct from the hdfs library isn't it? We'll have to live with that one!